lib/tom_queue/deferred_work_manager.rb

```diff
@@ -87,11 +86,10 @@ def start
       # (which have been scheduled by the AMQP consumer). If a message is returned
       # then we re-publish the messages to our internal QueueManager and ack the deferred
       # message
-      until @shutdown
+      while true
```

Comment:
Technically not, `while` doesn't have a nested scope for any variables created. Doesn't matter a huge amount here, but the `response`, `headers` and `payload` variables are left defined for the rest of the method outside the loop.

It's the same idiom as using `each` instead of `for`. Just because they have the same outcome doesn't mean they are equal in ranking for usage. Can we change this to be more rubyish please.